### DISCUSSION TITLE
How to write custom order with Prisma Client

### DISCUSSION THREAD

**Discussion by EBBach** [prisma/prisma#11816] [Q&A] [answered]
Hi, I would like to write a custom order with Prisma. Is there any ways to do it, or I have to work with Raw database access??

What I want to do is like 
```
SELECT * FROM Customers
ORDER BY case when country = 'P' then 1
      when country = 'A' then 2
      when country = 'L' then 3
      when country = 'H' then 4
      else 5
 end asc

```

Reference: https://stackoverflow.com/a/19196522

**Comment by Luke265** [comment, marked as answer]
Any query that involves `CASE`, `IF`, database functions, sub queries, window functions and other, a bit more complex features will not work with generated Prisma API. So you will have to use [$queryRaw](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access#queryraw) for this query.

There is an issue https://github.com/prisma/prisma/issues/5560, which would allow to write queries like this.

**Reply by EBBach** [reply]
Thank you for your answer, I will use $queryRaw.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
